shared/src/main/java/com/couchbase/lite/internal/support/StringUtils.java

```diff
@@ -0,0 +1,34 @@
+package com.couchbase.lite.internal.support;
+
+public class StringUtils {
+    // NSString - stringByDeletingLastPathComponent
+    // https://developer.apple.com/reference/foundation/nsstring/1411141-stringbydeletinglastpathcomponen
+    public static String stringByDeletingLastPathComponent(final String str) {
+        String path = str;
+        int start = str.length() - 1;
+        while (path.charAt(start) == '/')
```

Comment:
In case string ends with multiple forward slashes (`/`),  this loop ignores them.

For example, input str is `/tmp/lock//`